modules/light-clients/08-wasm/types/proposal_handle_test.go

```diff
@@ -78,6 +83,109 @@ func (suite *TypesTestSuite) TestCheckSubstituteAndUpdateStateGrandpa() {
 	}
 }
 
+func (suite *TypesTestSuite) TestCheckSubstituteAndUpdateState() {
+	var substituteClientState exported.ClientState
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			"success",
+			func() {
+				suite.mockVM.RegisterSudoCallback(
+					types.MigrateClientStoreMsg{},
+					func(_ cosmwasm.Checksum, _ wasmvmtypes.Env, sudoMsg []byte, store cosmwasm.KVStore, _ cosmwasm.GoAPI, _ cosmwasm.Querier, _ cosmwasm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+						var payload types.SudoMsg
+						err := json.Unmarshal(sudoMsg, &payload)
+						suite.Require().NoError(err)
+
+						suite.Require().NotNil(payload.MigrateClientStore)
+						suite.Require().Nil(payload.UpdateState)
+						suite.Require().Nil(payload.UpdateStateOnMisbehaviour)
+						suite.Require().Nil(payload.VerifyMembership)
+						suite.Require().Nil(payload.VerifyNonMembership)
+						suite.Require().Nil(payload.VerifyUpgradeAndUpdateState)
+
+						bz, err := json.Marshal(types.EmptyResult{})
+						suite.Require().NoError(err)
+
+						return &wasmvmtypes.Response{Data: bz}, wasmtesting.DefaultGasUsed, nil
+					},
+				)
+			},
+			nil,
+		},
+		{
+			"failure: invalid substitute client state",
+			func() {
+				substituteClientState = &ibctm.ClientState{}
+			},
+			clienttypes.ErrInvalidClient,
```

Comment:
could maybe wrap the error with the error string in the code to make it more specific, but fine as is

Reply:
could do, this is pretty prevalent currently. Could be nice to open overarching issue and maybe pick it up in future?

Reply:
I'm happy with whatever :)